### DISCUSSION TITLE
CORS with PAT when trying to access api from angular app

### DISCUSSION THREAD

**Discussion by mahi160** [firefly-iii/firefly-iii#6694] [General]
I am trying to create an ionic angular app. but after giving PAT as header, getting 
`No 'Access-Control-Allow-Origin' header is present on the requested resource.`

How can i use api with PAT then?
Using like this.
`
{Authorization: Bearer <PAT>}
`

Please help. TIA.

**Comment by JC5** [comment]
Where are you getting the error from exactly?

**Comment by mahi160** [comment]
thanks for the reply!
when ever i call an api, it shows that error.
here is the code.
![code](https://user-images.githubusercontent.com/50174250/205436820-f54997bf-a9ea-4763-8a18-4d355bd3ddab.png)

and this is the error
`auth:1 Access to XMLHttpRequest at 'http://192.168.0.103:8085/api/v1/about' from origin 'http://localhost:1234' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`
![image](https://user-images.githubusercontent.com/50174250/205436879-b9eaf20d-6d24-4d98-ac6c-04e84bdd31ae.png)

**Reply by JC5** [reply]
So the source website does not set the `Access-Control-Allow-Origin` header, which is something you must do in the reverse proxy or manually on the server running Firefly III. It's not part of the Docker image for example.

**Reply by mahi160** [reply]
Oh okay got it.
And if I use https I don't need to do this configs in proxy right? I mean user can connect easily?
Thanks!

**Reply by JC5** [reply]
You'll still need this, even when you're using https. The demo site has these headers as well.

**Reply by mahi160** [reply]
Okay thanks very much!
